src/vic.h

```diff
@@ -77,6 +77,9 @@
 extern char vic_buffer[VIC_BUFFER_SIZE + 1];
 extern uint8_t vic_buffer_len;
 
+#define vic_printf(format, ...) do { char _vic_buff[VIC_BUFFER_SIZE];\
```

Comment:
Ahoj,

Toto aby mi vzdy nezahrabalo stack som zvykol riesit tak ze som mal v
pameti jeden buffer cisto len na printovanie.
Ono to malo viac vyhod hentak.

Jo a velkost bufferu printovacieho som volil kolo 128-256, aj tak jeden
riadok vystupu by nemal byt nad 80 znakou.

co by som poradil je namiesto sprintf etc pouzite xitoa od takeho
cinanca, je to ovela mensie ako cele stdio a staci to. Silne doporucujem
tu kniznicu.

https://github.com/burlito/xitoa

On 02/20/2016 10:37 AM, Marek Šuppa wrote:
>> @@ -77,6 +77,9 @@
>>  extern char vic_buffer[VIC_BUFFER_SIZE + 1];
>>  extern uint8_t vic_buffer_len;
>>  
>> +#define vic_printf(format, ...) do { char _vic_buff[VIC_BUFFER_SIZE];\
> 
> @vargac I am wondering how well will this work in the real world: `VIC_BUFFER_SIZE` is pretty big and it might be that you will not often have enough space to conduct this whole operation.
> 
> Am I missing something obvious here? 
> 
> cc @burlito 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/xlcteam/vic/pull/24/files#r53552159
> 

-- 
Kind regards,
b.